src/compiler/transformers/ts.ts

```diff
@@ -1961,10 +1969,19 @@ namespace ts {
             //      this.y = y;
             //  }
             //
-            addRange(statements, map(parametersWithPropertyAssignments, transformParameterWithPropertyAssignment));
+            const parameterPropertyAssignments = mapDefined(parametersWithPropertyAssignments, transformParameterWithPropertyAssignment);
+
+            // If there is a super() call, the parameter properties go immediately after it
+            if (superStatementIndex !== undefined) {
+                addRange(statements, parameterPropertyAssignments);
+            }
+            // Since there was no super() call, parameter properties are the first statements in the constructor
+            else {
+                statements = addRange(parameterPropertyAssignments, statements);
+            }
 
-            // Add the existing statements, skipping the initial super call.
-            addRange(statements, visitNodes(body.statements, visitor, isStatement, indexOfFirstStatement));
+            // Add remaining statements from the body, skipping the super() call if it was found
+            addRange(statements, visitNodes(body.statements, visitor, isStatement, superStatementIndex === undefined ? 0 : superStatementIndex + 1));

```

Comment:
Another case where, if `findSuperStatementIndex` returns `-1` instead of `undefined`, this code could be simplified:

```suggestion
            addRange(statements, visitNodes(body.statements, visitor, isStatement, superStatementIndex + 1));
```

Reply:
There's something very aesthetically pleasing about how this happens to work out...